docs/compilers/CSharp/Compiler Breaking Changes - DotNet 10.md

```diff
@@ -291,3 +291,23 @@ unsafe record struct R(
     public bool Equals(R other) => true;
 }
 ```
+
+## Emitting metadata-only executables requires an entrypoint
+
+***Introduced in Visual Studio 2022 version 17.14***
+
+Previously, the entrypoint was [unintentionally unset](https://github.com/dotnet/roslyn/issues/76707)
+when emitting executables in metadata-only mode (also known as ref assemblies).
+That is now corrected but it also means that a missing entrypoint is a compilation error:
+
+```cs
+// previously successful, now fails:
+CSharpCompilation.Create("test").Emit(new MemoryStream(),
+    options: EmitOptions.Default.WithEmitMetadataOnly(true))
```

Comment:
>EmitOptions.Default.WithEmitMetadataOnly(true) [](http://example.com/codeflow?start=13&length=46)

Consider also adding an example that is using command line options. I think only small number of users work with the APIs directly.